File: THCS/2023_2024/QuangNinhB_2324/Bai2/pass.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>
using namespace std;

// Hàm tìm giá trị lớn nhất của biểu thức (ai - aj) * (ak - al)
long long findMaxPassword(const vector<int>& arr, int n) {
    // Sắp xếp mảng
    vector<int> sortedArr = arr;
    sort(sortedArr.begin(), sortedArr.end());

    // Tính hai lựa chọn
    long long option1 = (long long)(sortedArr[n - 1] - sortedArr[0]) * (sortedArr[n - 2] - sortedArr[1]);
    long long option2 = (long long)(sortedArr[n - 1] - sortedArr[1]) * (sortedArr[n - 2] - sortedArr[0]);

    // Kết quả là giá trị lớn nhất trong hai lựa chọn
    return max(option1, option2);
}

int main() {
    // Mở file input và output
    ifstream inputFile("pass.inp");
    ofstream outputFile("pass.out");

    // Đọc số phần tử
    int n;
    inputFile >> n;

    // Đọc mảng số nguyên
    vector<int> arr(n);
    for (int i = 0; i < n; ++i) {
        inputFile >> arr[i];
    }

    // Tìm kết quả và ghi ra file output
    long long result = findMaxPassword(arr, n);
    outputFile << result << endl;

    // Đóng file
    inputFile.close();
    outputFile.close();

    return 0;
}

File: THCS/2023_2024/QuangNinhB_2324/Bai2/pass.py
def find_max_password(n, arr):
    # Sắp xếp mảng
    arr.sort()
    
    # Chọn giá trị lớn nhất từ hai cặp hiệu
    # (a_n - a_1) * (a_{n-1} - a_2) và (a_n - a_2) * (a_{n-1} - a_1)
    option1 = (arr[-1] - arr[0]) * (arr[-2] - arr[1])
    option2 = (arr[-1] - arr[1]) * (arr[-2] - arr[0])
    
    # Kết quả là giá trị lớn nhất trong các lựa chọn
    return max(option1, option2)

# Đọc dữ liệu từ file
with open("pass.inp", "r") as f:
    n = int(f.readline())
    arr = list(map(int, f.readline().split()))

# Gọi hàm và tìm kết quả
result = find_max_password(n, arr)

# Ghi kết quả ra file
with open("pass.out", "w") as f:
    f.write(f"{result}\n")
